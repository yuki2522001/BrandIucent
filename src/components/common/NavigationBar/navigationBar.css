.nav {
  display: flex;
  position: relative;
  z-index: 20;
  justify-content: space-between;
  align-items: center;
  padding: 0 110px 0 162px;
}

.nav__about {
  display: flex;
  align-items: center;
  margin: 32px;
}