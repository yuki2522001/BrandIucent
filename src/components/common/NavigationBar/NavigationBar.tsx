import React, { memo } from "react";
import { MENU_LIST } from "@constants/menu";
import Logo from "../Logo/Logo";
import url from "../../../assets/images/logos/logo.png";
import Menu from "../Menu/Menu";
import Button from "../Button/Button";
import "./navigationBar.css";

interface NavigationBarProps {
  mode: "light" | "dark";
}

const NavigationBar: React.FC<NavigationBarProps> = memo(({ mode }) => {
  return (
    <nav data-testid="navigation-bar" className="nav">
      <Logo src={url} />
      <div className="nav__about">
        <Menu menuList={MENU_LIST} />
        {mode == "light" && (
          <>
            <Button text="Sign Up" type="light" />
            <Button text="Login" type="outline--light" />
          </>
        )}
        {mode =="dark" && (
          <>
            <Button text="Sign Up" type="dark" />
            <Button text="Login" type="outline--dark" />
          </>
        )}
      </div>
    </nav>
  );
});

export default NavigationBar;
