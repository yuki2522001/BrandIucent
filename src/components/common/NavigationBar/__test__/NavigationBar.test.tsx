import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import "@testing-library/jest-dom";
import NavigationBar from "../NavigationBar";

describe("Navigation component", () => {
  test("should render component NavigationBar type 'isThemeDark' ", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <NavigationBar mode="light" />
      </Router>
    );
    expect(getByTestId("navigation-bar")).toBeInTheDocument();
  });

  test("should render component NavigationBar type 'isThemeLight' ", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Router location={history.location} navigator={history}>
        <NavigationBar mode="dark" />
      </Router>
    );
    expect(getByTestId("navigation-bar")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Router location={history.location} navigator={history}>
        <NavigationBar mode="light" />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
