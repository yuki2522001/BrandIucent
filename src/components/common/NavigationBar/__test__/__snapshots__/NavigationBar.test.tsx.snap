// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="nav"
    data-testid="navigation-bar"
  >
    <a
      data-testid="logo"
      href="/"
    >
      <img
        alt="This is logo"
        class=""
      />
    </a>
    <div
      class="nav__about"
    >
      <div
        class="menu"
        data-testid="menu"
      >
        <ul
          class="menu__list"
        >
          <li
            class="menu__item"
            data-testid="menu-item"
          >
            <a
              href="/products"
            >
              Shop
            </a>
          </li>
          <li
            class="menu__item"
          >
            <a
              href="/"
            >
              Orders
            </a>
          </li>
          <li
            class="menu__item"
          >
            <a
              href="/"
            >
              Cart
            </a>
          </li>
          <li
            class="menu__item"
          >
            <a
              href="/"
            >
              Contact
            </a>
          </li>
        </ul>
      </div>
      <button
        class="btn btn--light"
      >
        Sign Up
      </button>
      <button
        class="btn btn__outline--light"
      >
        Login
      </button>
    </div>
  </nav>
</DocumentFragment>
`;
